File: on_thi_chuyen/CREATING_STRING.py
s = sorted(input())
n = len(s)
used = [False]*n
res = []
def backtrack(path):
    if len(path) == n:
        res.append("".join(path))
        return
    for i in range(n):
        if used[i]:
            continue
        if i>0 and s[i]==s[i-1] and not used[i-1]:
            continue
        used[i] = True
        path.append(s[i])
        backtrack(path)
        path.pop()
        used[i] = False
backtrack([])
print(len(res))
for x in res:
    print(x)

File: c_plus_plus/creating_string.cpp
#include <iostream>
#include <algorithm>
using namespace std;

string s;
int n;
bool used[8] = {false};
string res[40320];
int idx = 0;

void backtrack(string path){

    if(path.length() == n){
        res[idx++] = path;
        return;
    }

    for(int i = 0; i < n; i++){

        if(used[i]){
            continue;
        }

        if(i > 0 && s[i] == s[i-1] && !used[i-1]){
            continue;
        }

        used[i] = true;
        path += s[i];

        backtrack(path);

        path.pop_back();
        used[i] = false;
    }
}

int main(){

    cin >> s;

    sort(s.begin(), s.end());

    n = s.length();

    backtrack("");

    cout << idx << endl;

    for(int i = 0; i < idx; i++){
        cout << res[i] << endl;
    }
}
